coral-common/src/main/java/com/linkedin/coral/common/types/CoralTypeKind.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2024-2025 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.common.types;
+
+/**
+ * Enumeration of all supported data type kinds in the Coral type system.
+ * This provides a comprehensive set of primitive and complex types that
+ * can be mapped to various execution engines.
+ */
+public enum CoralTypeKind {
+  // Primitive numeric types
+  BOOLEAN,
```

Comment:
for backward compatibility (& a default mapping for unkown types), maybe we should also create NULL / OTHER as it exists today in [TypeConverter](https://github.com/linkedin/coral/blob/9f8dfce736e3b14aded7653cb340c75bf51dab7b/coral-common/src/main/java/com/linkedin/coral/common/TypeConverter.java#L99).

Reply:
I do not know if it is used in practice. Also coral conversion heavily relies on deterministic type inference. We can always add it in the future if the use case arises?